tenders/spt/spt_module_net.c

```diff
@@ -29,80 +29,124 @@
 #include <stdio.h>
 #include <string.h>
 #include <seccomp.h>
+#include <sys/epoll.h>
 
 #include "../common/tap_attach.h"
 #include "spt.h"
 
-static char *netiface;
-static int netfd = -1;
-static uint8_t guest_mac[6];
-static int cmdline_mac = 0;
+static bool module_in_use;
 
-static int handle_cmdarg(char *cmdarg)
+static int handle_cmdarg(char *cmdarg, struct mft *mft)
 {
-    if (!strncmp("--net=", cmdarg, 6)) {
-        netiface = cmdarg + 6;
-        return 0;
-    } else if (!strncmp("--net-mac=", cmdarg, 10)) {
-        const char *macptr = cmdarg + 10;
+    enum {
+        opt_net,
+        opt_net_mac
+    } which;
+
+    if (strncmp("--net:", cmdarg, 6) == 0)
+        which = opt_net;
+    else if (strncmp("--net-mac:", cmdarg, 10) == 0)
+        which = opt_net_mac;
+    else
+        return -1;
+
+    char name[MFT_NAME_SIZE];
+    char iface[20]; /* XXX should be IFNAMSIZ, needs extra header here */
+    int rc;
+    if (which == opt_net) {
+        rc = sscanf(cmdarg,
+                "--net:%" XSTR(MFT_NAME_MAX) "[A-Za-z0-9]="
+                "%19s", name, iface);
```

Comment:
wondered originally where this `19` came from, would it be possible to do `XSTR(sizeof(iface))` ?